### PR TITLE
[ARC][Seq] Switch arc over to use seq.clock

### PR DESCRIPTION
**Pull request by nandor**
This PR enforces the use of the clock type in some `arc` ops.

Part of #5851

### CODE REVIEW

File: lib/Conversion/SeqToSV/FirMemLowering.cpp [hunk 1]
```diff
@@ -226,7 +227,7 @@ FirMemLowering::createMemoryModule(FirMemConfig &mem,
   auto addCommonPorts = [&](StringRef prefix, size_t idx) {
     addInput(prefix, idx, "_addr", addrType);
     addInput(prefix, idx, "_en", bitType);
-    addInput(prefix, idx, "_clk", bitType);
+    addInput(prefix, idx, "_clk", clkType);
```
Comment (fabianschuiki) [inline lib/Conversion/SeqToSV/FirMemLowering.cpp:230]: Should this be part of #6053? There are a few other tweaks to Seq that look like it might belong to another PR.

Comment (nandor) [reply lib/Conversion/SeqToSV/FirMemLowering.cpp:230]: I don't think so - ARC detects memories and it's best if it sees a clock type after this pass instead of some clock conversions.

Comment (nandor) [reply lib/Conversion/SeqToSV/FirMemLowering.cpp:230]: I split off some stuff into this PR: https://github.com/llvm/circt/pull/6062

**Review by fabianschuiki — APPROVED** [review]
LGTM!
